### ISSUE TITLE
test requires STAN_NUM_THREADS but doesn't recognize it

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/math#1068]
## Description

Makefile for test `test/unit/math/prim/mat/functor/num_threads_test.cpp` doesn't work with default or with setting number of threads directly.  Here's what happens.  This is on Mac OSX High Sierra, with homebrew install of g++7.3.  Here's the `make/local`

```
CXX=/usr/local/Cellar/gcc@7/7.3.0/bin/g++-7
O=0
```

and here's where `g++-7` points:

```
~/cmdstan/stan/lib/stan_math(bugfix/1063-std-lgamma)$ /usr/local/Cellar/gcc@7/7.3.0/bin/g++-7 --version
g++-7 (Homebrew GCC 7.3.0) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

I clean, set the environment variable, then try to make and it fails because it can't find the environment variable.

```
~/cmdstan/stan/lib/stan_math(bugfix/1063-std-lgamma)$ make clean-all
  removing test executables
rm -f lib/gtest_1.8.1/src/gtest-all.o
  removing generated test files
rm -f 
  removing doxygen
rm -f -r doc/api
  removing dependency files
  cleaning sundials targets
rm -f 
```

```
~/cmdstan/stan/lib/stan_math(bugfix/1063-std-lgamma)$ export STAN_NUM_THREADS=2
```

```
~/cmdstan/stan/lib/stan_math(bugfix/1063-std-lgamma)$ ./runTests.py -j5 test/unit/math/prim/mat/functor/num_threads_test.cpp 
------------------------------------------------------------
make -j5 test/unit/math/prim/mat/functor/num_threads_test
/usr/local/Cellar/gcc@7/7.3.0/bin/g++-7 -std=c++1y       -O0  -I . -I lib/eigen_3.3.3 -I lib/boost_1.66.0 -I lib/sundials_3.1.0/include -I lib/gtest_1.8.1/include -I lib/gtest_1.8.1     -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION    -DGTEST_USE_OWN_TR1_TUPLE  -c -MT test/unit/math/prim/mat/functor/num_threads_test.o -MT test/unit/math/prim/mat/functor/num_threads_test.d -MM -E -MG -MP -MF test/unit/math/prim/mat/functor/num_threads_test.d test/unit/math/prim/mat/functor/num_threads_test.cpp
/usr/local/Cellar/gcc@7/7.3.0/bin/g++-7 -std=c++1y       -O0  -I . -I lib/eigen_3.3.3 -I lib/boost_1.66.0 -I lib/sundials_3.1.0/include -I lib/gtest_1.8.1/include -I lib/gtest_1.8.1 -I lib/gtest_1.8.1/include -I lib/gtest_1.8.1      -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION    -DGTEST_USE_OWN_TR1_TUPLE -DGTEST_USE_OWN_TR1_TUPLE  -c -o test/unit/math/prim/mat/functor/num_threads_test.o test/unit/math/prim/mat/functor/num_threads_test.cpp
/usr/local/Cellar/gcc@7/7.3.0/bin/g++-7 -std=c++1y       -O0  -I . -I lib/eigen_3.3.3 -I lib/boost_1.66.0 -I lib/sundials_3.1.0/include -I lib/gtest_1.8.1/include -I lib/gtest_1.8.1 -I lib/gtest_1.8.1/include -I lib/gtest_1.8.1      -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION    -DGTEST_USE_OWN_TR1_TUPLE -DGTEST_USE_OWN_TR1_TUPLE  -c -o lib/gtest_1.8.1/src/gtest-all.o lib/gtest_1.8.1/src/gtest-all.cc
/usr/local/Cellar/gcc@7/7.3.0/bin/g++-7 -std=c++1y       -O0  -I . -I lib/eigen_3.3.3 -I lib/boost_1.66.0 -I lib/sundials_3.1.0/include -I lib/gtest_1.8.1/include -I lib/gtest_1.8.1     -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION    -DGTEST_USE_OWN_TR1_TUPLE         test/unit/math/prim/mat/functor/num_threads_test.o lib/gtest_1.8.1/src/gtest_main.cc lib/gtest_1.8.1/src/gtest-all.o        -o test/unit/math/prim/mat/functor/num_threads_test
------------------------------------------------------------
test/unit/math/prim/mat/functor/num_threads_test --gtest_output="xml:test/unit/math/prim/mat/functor/num_threads_test.xml"
Running main() from lib/gtest_1.8.1/src/gtest_main.cc
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from num_threads
[ RUN      ] num_threads.correct_values
unknown file: Failure
C++ exception with description "get_num_threads(int): STAN_NUM_THREADS The STAN_NUM_THREADS environment variable is '' but it must be a positive number or -1" thrown in the test body.
[  FAILED  ] num_threads.correct_values (0 ms)
[ RUN      ] num_threads.incorrect_values
[       OK ] num_threads.incorrect_values (0 ms)
[----------] 2 tests from num_threads (0 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (0 ms total)
[  PASSED  ] 1 test.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] num_threads.correct_values

 1 FAILED TEST
test/unit/math/prim/mat/functor/num_threads_test --gtest_output="xml:test/unit/math/prim/mat/functor/num_threads_test.xml" failed
exit now (11/18/18 23:10:19 EST)


```

```
~/cmdstan/stan/lib/stan_math(bugfix/1063-std-lgamma)$ echo $STAN_NUM_THREADS
2
```

## Current Math Version
v2.18.0

**Comment by syclik** [comment]
@wds15, is this an easy fix?

**Comment by wds15** [comment]
maybe... not sure... need to check

**Comment by syclik** [comment]
Thanks. I’ve seen the code a while ago, but I’m not really familiar with it. It’ll take me a little bit just to find where this is. Hopefully you can just leave some notes behind if it’s a longer fix.

**Comment by wds15** [comment]
I think it's fixed... very weird!

**Comment by wds15** [comment]
fix was merged in #1070
